build-scripts/clang-tidy.sh

```diff
@@ -121,6 +121,19 @@ else
     fi
 fi
 
+printf "Subset to analyze: '%s'\n" "$CATA_CLANG_TIDY_SUBSET"
+
+# We might need to analyze only a subset of the files if they have been split
+# into multiple jobs for efficiency
+case "$CATA_CLANG_TIDY_SUBSET" in
+    ( src )
+        tidyable_cpp_files=$(printf '%s\n' "$tidyable_cpp_files" | grep '/src/')
+        ;;
+    ( other )
+        tidyable_cpp_files=$(printf '%s\n' "$tidyable_cpp_files" | grep -v '/src/')
```

Comment:
Now I know why. `grep` returns with exit code 1 if no file is matched, and this aborts the whole shell script.

Reply:
these PRs both changed files in the `src` folder though so it's strange for grep to fail in that case.